src/test/java/FreeTimeServletTest.java

```diff
@@ -0,0 +1,211 @@
+    Assert.assertEquals(500, actual.getWordCount());
+    Assert.assertEquals(50, actual.getAverageReadingSpeed());
+    Assert.assertEquals(10, actual.getMinutesToRead());
+    Assert.assertEquals(startHour.size(), actual.getPotentialEventTimes().size());
+    for (int index = 0; index < startHour.size(); index++) {
+      Assert.assertTrue(
+          startHour.get(index).equals(actual.getPotentialEventTimes().get(index).getStart()));
+      Assert.assertTrue(
+          endHour.get(index).equals(actual.getPotentialEventTimes().get(index).getEnd()));
+    }
+  }
+
+  @Test
+  public void alreadyCreatedEvent() throws Exception {
+    // Test case where there are unread emails that will take 10 minutes to read
+    // But an event (event 2) has already been created to go through them
+    // We should get no proposed potential event
+    Mockito.when(calendarClient.getCalendarList()).thenReturn(TWO_CALENDARS);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_ONE);
+    Mockito.when(calendarClient.getUpcomingEvents(SECONDARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_TWO);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    Mockito.when(gmailClient.getWordCount(7)).thenReturn(500);
+    PlanMailResponse actual = getServletResponse();
+    Assert.assertEquals(500, actual.getWordCount());
+    Assert.assertEquals(50, actual.getAverageReadingSpeed());
+    Assert.assertEquals(10, actual.getMinutesToRead());
+    Assert.assertTrue(actual.getPotentialEventTimes().isEmpty());
+  }
+
+  @Test
+  public void splitPotentialEvent() throws Exception {
+    // Test case where there are unread emails that will take 10 minutes to read
+    // and an event that splits the time interval required in two parts
+    // We should get 2 proposed events of 10 minutes each.
+    Mockito.when(calendarClient.getCalendarList()).thenReturn(ONE_CALENDAR);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(EVENT_THREE);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    Mockito.when(gmailClient.getWordCount(7)).thenReturn(500);
+    PlanMailResponse actual = getServletResponse();
+    List<Date> startHour =
+        Arrays.asList(
+            new Date(2020 - OFFSET_YEAR, 4, 19, 10, 0),
+            new Date(2020 - OFFSET_YEAR, 4, 19, 10, 30));
+    List<Date> endHour =
+        Arrays.asList(
+            new Date(2020 - OFFSET_YEAR, 4, 19, 10, 5),
+            new Date(2020 - OFFSET_YEAR, 4, 19, 10, 35));
+    Assert.assertEquals(500, actual.getWordCount());
+    Assert.assertEquals(50, actual.getAverageReadingSpeed());
+    Assert.assertEquals(10, actual.getMinutesToRead());
+    Assert.assertEquals(startHour.size(), actual.getPotentialEventTimes().size());
+    for (int index = 0; index < startHour.size(); index++) {
```

Comment:
Instead of this loop, consider overriding the `public int hashCode()` and `public boolean equals()` methods to determine equality. Take a look at `ActionableMessage`, as I did this there with some helpful Apache tools

You can then just use `Assert.assertEquals(expectedPlanMailResponse, actualPlanMailResponse)`, which is more readable. 

Don't spend too much time on this unless if @gbuenoandrade or @antmar thinks this is a high-value thing to do, but this might help clean things up and avoid using a loop for assertions.

Reply:
I overrode the equal method in `PlanMailResponse`, `Resolved`

Reply:
Cool! Thanks